satellite/metainfo/metainfo.go

```diff
@@ -514,7 +517,21 @@ func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Poi
 		}
 		signee := signing.SigneeFromPeerIdentity(peerID)
 
-		err = endpoint.validatePieceHash(ctx, piece, limits, signee)
+		limit := originalLimits[piece.PieceNum]
```

Comment:
Should we check that the pieceNum is valid before we try to retrieve the limit? An invalid piece num may cause a panic here